### PR TITLE
Added "Continue" link to error page (#40909)

### PR DESCRIPTION
**Pull request by mijarena**
## Related Issue

- #40909

## Motivation and Context
User can quickly return to the starting page when encountering an error.

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by jnweiger — CHANGES_REQUESTED** [review]
href="/" is not always correct.

There are two config variables that need to be taken into account:
```
/**
 * Define the Web base URL
 * This key is necessary for the navigation item to the new ownCloud Web UI and for redirecting
 * public and private links.
 */
'web.baseUrl' => '',
/**
 * Define clean URLs without `/index.php`
 * This parameter will be written as `RewriteBase` on update and installation of
 * ownCloud to your `.htaccess` file. While this value is often simply the URL
 * path of the ownCloud installation it cannot be set automatically properly in
 * every scenario and needs thus some manual configuration.
 *
 * In a standard Apache setup this usually equals the folder that ownCloud is
 * accessible at. So if ownCloud is accessible via `https://mycloud.org/owncloud`
 * the correct value would most likely be `/owncloud`. If ownCloud is running
 * under `https://mycloud.org/` then it would be `/`.
 *
 * Note that the above rule is not valid in every case, as there are some rare setup
 * cases where this may not apply. However, to avoid any update problems this
 * configuration value is explicitly opt-in.
 *
 * After setting this value run `{occ-command-example-prefix} maintenance:update:htaccess`. Now, when the
 * following conditions are met ownCloud URLs won't contain `index.php`:
 *
 * - `mod_rewrite` is installed
 * - `mod_env` is installed
 */
'htaccess.RewriteBase' => '/',
```

ownCloud may also be installed like this:

https://oc10130a1-20230807.jw-qa.owncloud.works/mammamia/owncloud/login
config.php: `'htaccess.RewriteBase' => '/mammamia/owncloud',`

https://oc1012-index-20230807.jw-qa.owncloud.works/index.php
No RewriteBase in config.php

https://oc10122-both-20230807.jw-qa.owncloud.works/big-bang/index.php
apache.conf: `Alias /big-bang "/var/www/owncloud/"` without a RewriteBase in config.php


https://oc10130a1-web-app-20230807.jw-qa.owncloud.works/index.php
... and what happens with the new web UI? (Optional, but it should not break...)

**Comment by mijarena** [comment]
Fixed on 403 page, too

**Review by jnweiger — APPROVED** [review]
Work in all my test cases.
* admin permission, wrong file id
* index.php less and other
* / and subfolder install.

**Comment by IljaN** [comment]
- Should we make the "Continue" prettier. Maybe under the message box? But fine for me.
- CI is failing https://drone.owncloud.com/owncloud/core/38908/3/7
- Please add a changelog entry https://github.com/owncloud/core/pull/40910#issuecomment-1667830427
- Added additional reviewers since we worked on the code together ;)

**Review by IljaN — CHANGES_REQUESTED** [review]
- Should we make the "Continue" prettier. Maybe under the message box? But fine for me.
- CI is failing https://drone.owncloud.com/owncloud/core/38908/3/7
- Please add a changelog entry https://github.com/owncloud/core/pull/40910#issuecomment-1667830427
- Added additional reviewers since we worked on the code together ;)

**Comment by mijarena** [comment]
I made this to be working correctly So it could be an accepted feature for the next release. I planed to polish code, style and translation next during beta.

File: apps/files/lib/Controller/ViewController.php [hunk 1]
```diff
@@ -335,7 +335,17 @@ public function showFile($fileId, $details = null) {
...
 		if ($this->userSession->isLoggedIn() and empty($files)) {
 			$param["error"] = $this->l10n->t("You don't have permissions to access this file/folder - Please contact the owner to share it with you.");
-			$response = new TemplateResponse("core", 'error', ["errors" => [$param]], 'guest');
+
+			$response = new TemplateResponse(
+				"core",
+				'error',
+				[
+					"errors" => [$param],
+					"base_url" => \OC::$server->getURLGenerator()->getAbsoluteURL("/"),
```
Comment (jvillafanez) [inline apps/files/lib/Controller/ViewController.php:344]: Is it possible to inject the `URLGenerator`?

**Review by jvillafanez — COMMENTED** [review]
Not sure if it's easy to add some basic unit tests.

**Comment by DeepDiver1975** [comment]
cleaning up - I think this is no longer of anybody's interest

**Pull request closed without merging**